infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -67,49 +87,206 @@ public class LogReplicationConfigManager {
+     */
+    public boolean loadRegistryTableEntries() {
+        try {
+            log.info("Sync in-memory registry table entries with retry.");
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try {
+                    StreamAddressSpace currentAddressSpace = rt.getSequencerView().getStreamAddressSpace(
+                            new StreamAddressRange(REGISTRY_TABLE_ID, Long.MAX_VALUE, Address.NON_ADDRESS));
+                    if (lastAddressSpace == null || !lastAddressSpace.equals(currentAddressSpace)) {
+                        lastAddressSpace = currentAddressSpace;
+                        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
+                                rt.getTableRegistry().getRegistryTable();
+                        registryTableEntries = registryTable.entryStream().collect(Collectors.toList());
+                        return true;
+                    }
+                } catch (Exception e) {
+                    log.error("Exception caught when syncing with registry table, retry needed", e);
+                    throw new RetryNeededException();
+                }
+                return false;
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            log.error("Cannot sync with registry table, LR stopped", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Read registry table's entries up to the given timestamp into memory for config synchronization.
+     *
+     * @param timestamp the timestamp up to which registry table should be synced.
+     */
+    public void loadRegistryTableEntriesAtSnapshot(long timestamp) {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                CorfuStoreMetadata.Timestamp snapshotTimestamp = CorfuStoreMetadata.Timestamp.newBuilder()
+                        .setEpoch(rt.getLayoutView().getLayout().getEpoch())
+                        .setSequence(timestamp)
+                        .build();
+                log.info("Sync in-memory registry table entries with timestamp: {}", snapshotTimestamp);
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE, IsolationLevel.snapshot(snapshotTimestamp))) {
+                    CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
+                            rt.getTableRegistry().getRegistryTable();
+                    registryTableEntries = registryTable.entryStream().collect(Collectors.toList());
+                    txn.commit();
+                } catch (Exception e) {
+                    throw new RetryNeededException();
+                }
+                // Reset lastAddressSpace to enforce next re-sync with registry table.
+                lastAddressSpace = null;
```

Comment:
I looked around and seems like we can get the streamAddressSpace at a ts. 
https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java#L317
the upper bound can be the snapshotTs.

if we can have this, it will be a small optimization where we will not mandatorily re-sync after a snapshot sync.